### PR TITLE
build: split build and blobs by architecture

### PR DESCRIPTION
**Pull request by alcioa**
Split the blobs and examples into their own directories
by architecture and add the aarch64 Dockerfile.

Builds on aarch64 machine. No cross build support/testing.

Signed-off-by: Alexandru Ciobotaru <alcioa@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,11 @@
 
 .DEFAULT_GOAL := nitro-cli
 
+ARCH_x86_64      = x86_64
```
Comment (alexggh) [inline Makefile:11]: pull requests checks are failing, could you have a look to see what is going on ?

Comment (alcioa) [reply Makefile:11]: Sorry, forgot to mention yesterday. The bump to Rust 1.49.0 bump shall cause `make nitro-clippy` and `make nitro-format` to fail. We shall fix the 1.49.0 issues in a separate PR and this PR should pass after I'll rebase and repush.

**Review by alexggh — APPROVED** [review]
Looks good to me.
Why did we change the sample blobs ?

**Comment by alcioa** [comment]
> Looks good to me.
> Why did we change the sample blobs ?

The sample blobs were symbolic links to the root directory blobs and they had to be reconstructed with the new per-arch directory layout.

**Review by andraprs — DISMISSED** [review]
May need also an update to default blobs path, based on arch options. Should  double-check the RPM spec, if all good with blobs setup at the given path.

https://github.com/aws/aws-nitro-enclaves-cli/blob/main/src/lib.rs#L41

**Comment by alcioa** [comment]
> May need also an update to default blobs path, based on arch options. Should double-check the RPM spec, if all good with blobs setup at the given path.
> 
> https://github.com/aws/aws-nitro-enclaves-cli/blob/main/src/lib.rs#L41

I initially took a look at that path. That's the same for each architecture, i.e. where we copy these blobs to that path. So for both archs we now copy them from the `blobs/<arch>/` to `/usr/share/nitro-enclaves`. Had a chat cu bercarug@ so https://github.com/aws/aws-nitro-enclaves-cli/blob/main/SPECS/aws-nitro-enclaves-cli.spec#L186 is the post-install `/usr/share/nitro-enclaves/blobs` path which is right now common for both archs. Of course, we do not have ARM RPM spec right now. Thanks.

**Comment by andraprs** [comment]
> > May need also an update to default blobs path, based on arch options. Should double-check the RPM spec, if all good with blobs setup at the given path.
> > https://github.com/aws/aws-nitro-enclaves-cli/blob/main/src/lib.rs#L41
> 
> I initially took a look at that path. That's the same for each architecture, i.e. where we copy these blobs to that path. So for both archs we now copy them from the `blobs/<arch>/` to `/usr/share/nitro-enclaves`. Had a chat cu bercarug@ so https://github.com/aws/aws-nitro-enclaves-cli/blob/main/SPECS/aws-nitro-enclaves-cli.spec#L186 is the post-install `/usr/share/nitro-enclaves/blobs` path which is right now common for both archs. Of course, we do not have ARM RPM spec right now. Thanks.

Ok, thanks for checking it out.

**Review by andraprs — APPROVED** [review]

**Review by alexggh — APPROVED** [review]

**Pull request merged**